### DISCUSSION TITLE
Getting Started Test Drive is not accurate for current template.

### DISCUSSION THREAD

**Discussion by IvanTheGeek** [AvaloniaUI/Avalonia#13578] [Q&A] [answered]
I am investigating Avalonia and come from a highly technical background but not a programming one. 
I installed Visual Studio Community and the Avalonia plugin which provides the project templates. 
I found the [Getting Started](https://docs.avaloniaui.net/docs/next/get-started/test-drive/) section, which do not match the options listed in the getting started. 

I am trying to go through the [test drive](https://docs.avaloniaui.net/docs/next/get-started/test-drive/) section.  Again, the file setups in the template do not match the instructions. I would normally not be so picky, but this is specifically for newbies. I have tried to add the `<Button>` tag, but can not find anywhere to put it that compiles!

Based on the current template, where can I make changes?? I am willing to work on updating the guides, but I can not do that until i even understand how that template is setup and where to start changing it!

**Comment by thevortexcloud** [comment, marked as answer]
Some of Avalonia's docs have not been updated in awhile and _may_ not work any more. Can you tell me which part you are stuck on and what you have so far? 

You should just need to add it into your XAML (Likely called something like `MainWindow.axaml`). Based on where _I_ think you are in the guide, I am guessing it's this page:

https://docs.avaloniaui.net/docs/next/get-started/test-drive/add-a-control

I can't see anything wrong with that page specifically though. But as a guess as what is wrong: Windows can only contain one _direct_ child. As such if you want to only display the button in the window you need to remove everything else. Generally to get around this you use some sort of layout panel such as a `Grid`/`Panel`/`DockPanel`/etc. The guide does show this correctly but it does not really explain it at that point.

**Reply by IvanTheGeek** [reply]
Yes, thank you. It took me a while, but I finally got the point of only one direct child in the window tag. By skipping around in some samples, I was able to find a piece of code where I replaced the whole window code and get it working. 
I will probably branch the repo and do some updating when I have a chance.

**Reply by thevortexcloud** [reply]
I am confused what needs to be updated here. At that point in the guide it does not tell you anything about using a second control. It does however explain how to do it on the very next page and even mentions this restriction.

> In fact an Avalonia window allows only a single other control in its content zone. So to add the multiple visual elements required for most applications, you will need to use one of the built-in layout controls that allows multiple other controls inside its content zone.

The only thing I can think of would be to change it to show the full Window XAML so people who skip through it can see it better.
